docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,295 @@
+$ oras attach --artifact-type oras/test localhost:5000/command/images:foobar --distribution-spec v1.0
+Error: unknown distribution specification flag: v1.0
+```
+
+Suggested error message:
+
+```console
+$ oras attach --artifact-type oras/test localhost:5000/sample/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "v1.0". 
+Available options: v1.1-referrers-api, v1.1-referrers-tag
+```
+
+#### Example 6: When attaching, if neither file reference nor annotation is provided
+
+Current behavior and output:
+
+```console
+$ oras attach --artifact-type sbom/example localhost:5000/sample/images:foobar
+Error: no blob is provided
+```
+
+Suggested error message:
+
+```console
+$ oras attach --artifact-type sbom/example localhost:5000/sample/images:foobar
+Error: neither file nor annotation provided in the command
+Usage: oras attach [flags] --artifact-type=<type> <name>{:<tag>|@<digest>} <file>[:<type>] [...]
+To attach to an existing artifact, please provide files via argument or annotations via flag "--annotation". Run "oras attach -h" for more options and examples
+```
+
+#### Example 7: When pushing files, if the annotation file doesn't match the required format
+
+Current behavior and output:
+
+```console
+$ oras push --annotation-file sbom.json ghcr.io/library/alpine:3.9
+Error: failed to load annotations from sbom.json: json: cannot unmarshal string into Go value of type map[string]map[string]string. Please refer to the document at https://oras.land/docs/how_to_guides/manifest_annotations
+```
+
+Suggested error message:
+
+```console
+$ oras push --annotation-file annotation.json ghcr.io/library/alpine:3.9
+Error: invalid annotation json file: failed to load annotations from annotation.json.
+Annotation file doesn't match the required format. Please refer to the document at https://oras.land/docs/how_to_guides/manifest_annotations
+```
+
+#### Example 8: When pushing files, if the annotation value doesn't match the required syntax
+
+Current behavior and output:
+
+```console
+$ oras push --annotation "key:value" ghcr.io/library/alpine:3.9
+Error: missing key in `--annotation` flag: key:value
+```
+
+Suggested error message:
+
+```console
+$ oras push --annotation "key:value" ghcr.io/library/alpine:3.9
+Error: annotation value doesn't match the required format.
+Please use the correct format in the flag: --annotation "key=value"  
+```
+
+#### Example 9: When failed to pull files from a public registry
+
+```console
+$ oras pull docker.io/nginx:latest
+Error: failed to resolve latest: GET "https://registry-1.docker.io/v2/nginx/manifests/latest": response status code 401: unauthorized: authentication required: [map[Action:pull Class: Name:nginx Type:repository]]
+```
+
+Suggested error message:
+
+```console
+$ oras pull docker.io/nginx:latest
+Error response from registry: pull access denied for docker.io/nginx:latest : response status code 401: unauthorized: requested access to the resource is denied
```

Comment:
Should not output status code as registry already returns error code, message and detail.
```suggestion
Error response from registry: pull access denied for docker.io/nginx:latest : unauthorized: requested access to the resource is denied
```